lib/dialects/abstract/query.js

```diff
@@ -514,7 +514,19 @@ AbstractQuery.prototype.isUpdateQuery = function() {
 
 AbstractQuery.prototype.handleSelectQuery = function(results) {
   var result = null;
-
+  // Map raw fields to arbitrary names if a mapping is provided
+  if (this.options.fieldMap) {
```

Comment:
I assumed it did this by design as handling a `UPDATE ... RETURNING *` is similar to a select. Here is the call to `handleSelectQuery` in [postgres/query.js](https://github.com/sequelize/sequelize/blob/eaf197fa7e6cb4d6b52d3d13493e6ae81352fdf8/lib/dialects/postgres/query.js#L296).